homeassistant/scripts/check_config.py

```diff
@@ -165,7 +190,11 @@ def run(script_args: list) -> int:
                 continue
             print(" -", skey + ":", sval)
 
-    return len(res["except"])
+    # Determine final exit code
+    exit_code = len(res["except"])
+    if args.fail_on_warnings and res["warn"]:
+        exit_code = max(exit_code, 1)
+    return exit_code
```

Comment:
We can probably simplify this (and above), as a non-zero return code indicates an error, rather than trying to indicate a count of errors. 

```suggestion
    return len(res["except"]) + int(args.fail_on_warnings and res["warn"])
```